taskboot/git.py

```diff
@@ -130,3 +131,35 @@ def github_release(target, args):
         release.upload_asset(name=asset_name, path=artifact_path, label=asset_name)
 
     logger.info(f"Release available as {release.html_url}")
+
+
+def github_push(args):
+
+    # Load config from file/secret
+    config = Configuration(args)
+    assert config.has_github_auth(), "Missing GitHub authentication"
+
+    # Set remote repository
+    repo_link = "https://{}:{}@github.com/{}.git".format(
+        args.user, config.github["token"], args.repository
+    )
+    subprocess.run(["git", "remote", "set-url", "origin", repo_link])
+
+    # Push on repository
+    try:
+        if args.force_push:
+            subprocess.run(["git", "push", "-f", "origin", args.branch], check=True)
+        else:
+            subprocess.run(["git", "push", "origin", args.branch], check=True)
+
+    except subprocess.CalledProcessError:
+        raise Exception(f"Pushing commits to {args.repository} failed")
+
+
+def git_push(target, args):
+    """
+    Push commits on a repository
+    """
+
+    # Push commits on a GitHub repository
+    github_push(args)
```

Comment:
I'd put the code directly in `git_push`, it doesn't have to be GitHub-specific.

Reply:
The main problem is `assert config.has_github_auth(), "Missing GitHub authentication"`. For this reason I have splitted in a separate function.

Reply:
Eventually we could add a function for `Gitlab`, `BitBucket` and so on just adding a new argument that specifies the hosting

Reply:
> The main problem is assert config.has_github_auth(), "Missing GitHub authentication". For this reason I have created a separate function.

We can change that in a follow-up. `has_github_auth` is just a name, but in practice it can also be a password for any other git service.

Reply:
> Eventually we could add a function for Gitlab, BitBucket and so on and just adding a new argument to discriminate the function to use

The functions for all git services would be the same, so I don't think it's worth it to have them be separate. We should just have one which is used for any git service.

Reply:
Ok, let's maintain that check for now, I'm going to change that in a follow-up PR. I think the `github.com` part shouldn't be added to the repository argument, but it should be a different argument.